src/Stratis.Bitcoin.Features.Interop/ETHClient/ContractSource/WrappedToken.sol

```diff
@@ -0,0 +1,60 @@
+// contracts/WrappedToken.sol
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.6.0;
+
+import "https://github.com/OpenZeppelin/openzeppelin-contracts/blob/v3.3.0/contracts/access/Ownable.sol";
+import "https://github.com/OpenZeppelin/openzeppelin-contracts/blob/v3.3.0/contracts/token/ERC20/ERC20.sol";
+
+contract WrappedToken is ERC20, Ownable {
+    mapping (address => string) public withdrawalAddresses;
+    
+    constructor(string tokenName, string tokenSymbol, uint256 initialSupply) public ERC20(tokenName, tokenSymbol) {
+        _mint(msg.sender, initialSupply);
+    }
+    
+    /**
+     * @dev Creates `amount` new tokens and assigns them to `account`.
+     *
+     * See {ERC20-_mint}.
+     */
+    function mint(address account, uint256 amount) public onlyOwner {
+        _mint(account, amount);
+    }
+    
+    /**
+     * @dev Destroys `amount` tokens from the caller.
+     *
+     * See {ERC20-_burn}.
+     */
+    function burn(uint256 amount, string memory address) public {
+        _burn(_msgSender(), amount);
+        
+        // When the tokens are burnt we need to know where to credit the equivalent value on the Stratis chain.
+        // Currently it is only possible to assign a single address here, so if multiple recipients are required
+        // the burner will have to wait until each burn is processed before proceeding with the next.
+        withdrawalAddresses[msg.sender] = address;
```

Comment:
maybe add additional method with onlyOwner modifier that removes value withdrawalAddresses[address]. 
And in burn method if address is set- revert. 
That will prevent users sending requests with multiple recepients in a row before prev requests are processed and subsequently loosing money

Reply:
@noescape00 , I've made the following change:
```
        string memory key = string(abi.encodePacked(msg.sender, " ", burnId));
        withdrawalAddresses[key] = tokenAddress;
```